fdbrpc/ContinuousSample.h

```diff
@@ -66,7 +66,9 @@ class ContinuousSample {
 		if (!samples.size() || percentile < 0.0 || percentile > 1.0)
 			return T();
 		sort();
-		int idx = std::floor((samples.size() - 1) * percentile);
+		// Percentile (X) is the smallest element in the sample set at least as large as X% of the samples.
+		// Min with last idx to ensure no out of bounds for percentile=1.0
+		int idx = std::min((int)std::ceil(samples.size() - 1 * percentile), (int)samples.size() - 1);
```

Comment:
I think this should be:

```suggestion
		int idx = std::min<int>(samples.size() * percentile, samples.size() - 1);
```

As written (assuming that we correct the order of operations issue), the code would still have N-1 discrete buckets while I think you actually want N buckets.

For example, with the set [1,2,3,4], my proposal would return 1 on the interval [0, 0.25), 2 on the interval [0.25, 0.5), 3 on the interval [0.5, 0.75), and 4 on the interval [0.75, 1.0].

The above formulation would return 1 on the interval [0, 0], 2 on the interval (0, 1/3], 3 on the interval (1/3, 2/3], and 4 on the interval (2/3, 1], and this would be a mirrored version of the existing logic.

Reply:
Technically, the way I wrote it would not match up with the description exactly. Instead, PXX would be the smallest element in the set that is larger than (but not exactly) XX% of samples, with the exception of P100. I don't think the difference here matters much, but writing a version that matches exactly your description would look like this and is maybe a little more opaque:

`int idx = std::max<int>(std::ceil(samples.size() * percentile) - 1, 0);`

Then you would get 1 on the interval [0, 0.25], 2 on the interval (0.25, 0.5], 3 on the interval (0.5, 0.75], and 4 on the interval (0.75, 1.0].

Reply:
Yeah I talked with steve about this independently as well and the last one you mentioned I think makes the most sense.